### PR TITLE
Adding gRPC query for escrow address

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #840 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by AdityaSripal — APPROVED** [review]
Nice work!

File: proto/ibc/applications/transfer/v1/query.proto [hunk 1]
```diff
@@ -30,6 +30,11 @@ service Query {
   rpc DenomHash(QueryDenomHashRequest) returns (QueryDenomHashResponse) {
     option (google.api.http).get = "/ibc/apps/transfer/v1/denom_hashes/{trace}";
   }
+
+  // EscrowAddress returns the escrow address for a particular port and channel id.
+  rpc EscrowAddress(QueryEscrowAddressRequest) returns (QueryEscrowAddressResponse) {
+    option (google.api.http).get = "/ibc/apps/transfer/v1/escrow_address/port/{port_id}/channel/{channel_id}";
```
Comment (crodriguezvega) [inline proto/ibc/applications/transfer/v1/query.proto:36]: Sorry, @chatton, I just realized that maybe the REST url should be different (hence my comment on Slack as well)...

```suggestion
    option (google.api.http).get = "/ibc/apps/transfer/v1/channels/{channel_id}ports/{port_id}/escrow_address";
```

**Review by crodriguezvega — CHANGES_REQUESTED** [review]
Awesome! 🎸 
Let's discuss and decide if we should adjust the REST url...

File: proto/ibc/applications/transfer/v1/query.proto [hunk 2]
```diff
@@ -84,3 +89,14 @@ message QueryDenomHashResponse {
...
+// QueryEscrowAddressRequest is the request type for the EscrowAddress RPC method.
+message QueryEscrowAddressRequest {
+  string port_id    = 1;
+  string channel_id = 2;
+}
+
+// QueryEscrowAddressResponse is the response type of the EscrowAddress RPC method.
+message QueryEscrowAddressResponse {
+  string escrow_address = 1;
+}
```
Comment (damiannolan) [inline proto/ibc/applications/transfer/v1/query.proto:102]: Could we add some protodoc comments for the message fields? Just basic standard ones we have in other places. 
Something like:

```
// unique port identifier
string port_id = 1;
// unique channel identifier
string channel_id = 2;

---

// the escrow account address
string escrow_address = 1;
```

**Review by damiannolan — APPROVED** [review]
Nice work!

**Comment by crodriguezvega** [comment]
We need also update the swagger.yaml file. You need to run `make proto-swagger-gen` locally and it will update the file.

File: proto/ibc/applications/transfer/v1/query.proto (hunk 2 above)
Comment (chatton) [reply proto/ibc/applications/transfer/v1/query.proto:102]: makes sense will add some comments!

**Pull request merged**